src/Rules/Password.php

```diff
@@ -66,14 +77,39 @@ public function message()
             return $this->message;
         }
 
-        if ($this->requireUppercase && ! $this->requireNumeric) {
-            return __('The :attribute must be at least '.$this->length.' characters and contain at least one uppercase character.');
-        } elseif ($this->requireNumeric && ! $this->requireUppercase) {
-            return __('The :attribute must be at least '.$this->length.' characters and contain at least one number.');
-        } elseif ($this->requireUppercase && $this->requireNumeric) {
-            return __('The :attribute must be at least '.$this->length.' characters and contain at least one uppercase character and number.');
-        } else {
-            return __('The :attribute must be at least '.$this->length.' characters.');
+        switch (1) {
+            case $this->requireUppercase
+                && ! $this->requireNumeric
+                && ! $this->requireSpecialCharacter:
+                return __('The :attribute must be at least '.$this->length.' characters and contain at least one uppercase character.');
```

Comment:
It should probably changed to something like: `return __('The :attribute must be at least :length characters and contain at least one uppercase character.', ['length' => $this->length]);`

Reply:
Please check it now

Reply:
@leo95batista looks better!